pkg/api/v1/atlasdatabaseuser_types.go

```diff
@@ -51,9 +51,12 @@ const (
 )
 
 // AtlasDatabaseUserSpec defines the desired state of Database User in Atlas
+// +kubebuilder:validation:XValidation:rule="has(self.atlasRef) && !has(self.projectRef) ? true : !has(self.atlasRef) && has(self.projectRef)",message="cannot define both atlasRef and projectRef at same time"
```

Comment:
nit: it can be shortened a bit: 
`// +kubebuilder:validation:XValidation:rule="(!has(self.atlasRef) && has(self.projectRef)) || (has(self.atlasRef) && !has(self.projectRef))`